web/packages/teleterm/src/ui/DocumentGatewayApp/AppGateway.tsx

```diff
@@ -118,3 +160,86 @@ export function AppGateway(props: {
     </Box>
   );
 }
+
+const LabelWithAttemptStatus = (props: {
+  text: string;
+  attempt: Attempt<unknown>;
+}) => (
+  <Flex
+    alignItems="center"
+    justifyContent="space-between"
+    css={`
+      position: relative;
+    `}
+  >
+    {props.text}
+    {props.attempt.status === 'processing' && (
+      <AnimatedSpinner color="interactive.tonal.neutral.2" size="medium" />
+    )}
+    {props.attempt.status === 'success' && (
+      // CSS animations are repeated whenever the parent goes from `display: none` to something
+      // else. As a result, we need to unmount the animated check so that the animation is not
+      // repeated when the user switches to this tab.
+      // https://www.w3.org/TR/css-animations-1/#example-4e34d7ba
+      <UnmountAfter timeoutMs={disappearanceDelayMs + disappearanceDurationMs}>
+        <DisappearingCheck color="success.main" size="medium" />
+      </UnmountAfter>
```

Comment:
Nit: IMO these icons better align with the labels when they are small instead of medium.